contracts/RewardsPoolFactory.sol

```diff
@@ -0,0 +1,167 @@
+     */
+    function startStakings() external {
+        require(rewardsPools.length != 0, 'RewardsPoolFactory::startStakings: called before any deploys');
+
+        for (uint256 i = 0; i < rewardsPools.length; i++) {
+            startStaking(rewardsPools[i]);
+        }
+    }
+
+    /** @dev Function to determine whether the staking and rewards distribution has stared for a given RewardsPool contract
+     * @param rewardsPoolAddress The address of the reward pools contract
+     */
+    function hasStakingStarted(address rewardsPoolAddress)
+        public
+        view
+        returns (bool)
+    {
+        return RewardsPool(rewardsPoolAddress).hasPeriodStarted();
+    }
+
+    /** @dev Starts the staking and rewards distribution for a given staking token. This is a fallback in case the startStakings() costs too much gas to call for all contracts
+     * @param rewardsPoolAddress The address of liquidity mining campaign
+     */
+    function startStaking(address rewardsPoolAddress) public {
+        require(block.timestamp >= rewardsPoolGenesis, 'RewardsPoolFactory::startStaking: not ready');
+        require(rewardsPoolAddress != address(0), 'RewardsPoolFactory::startStaking: not deployed');
+
+
+        RewardsPool srInstance = RewardsPool(rewardsPoolAddress);
+       
+        require(!hasStakingStarted(rewardsPoolAddress), 'RewardsPoolFactory::startStaking: Staking has started');
+
+        uint256 rtsSize = srInstance.getRewardsTokensCount();
+        for (uint256 i = 0; i < rtsSize; i++) {
+            require(
+                IERC20Detailed(srInstance.rewardsTokensArr(i))
+                    .transfer(rewardsPoolAddress, srInstance.rewardsAmountsArr(i)),
+                'RewardsPoolFactory::startStaking: transfer failed'
+            );
+        }
+
+        srInstance.start();
+    }
+
+    /** @dev Triggers the withdrawal of LP rewards from the rewards pool contract to the given recipient address
+     * @param rewardsPoolAddress The address of the token being staked
+     * @param recipient The address to whom the rewards will be trasferred
+     * @param lpTokenContract The address of the rewards contract
+     */
+    function withdrawLPRewards(address rewardsPoolAddress, address recipient, address lpTokenContract)
+        external
+        onlyOwner {
+
+        require(rewardsPoolAddress != address(0), 'RewardsPoolFactory::startStaking: not deployed');
+        RewardsPool srInstance = RewardsPool(rewardsPoolAddress);
+        srInstance.withdrawLPRewards(recipient, lpTokenContract);
+        
+    }
+
+    /** @dev Triggers the withdrawal of rewards from the rewards pool contract.
+     * @param rewardsPoolAddress The address of the token being staked
+     */
+    function withdrawRewards(address rewardsPoolAddress)
+        external
+        onlyOwner {
+            
+        require(rewardsPoolAddress != address(0), 'RewardsPoolFactory::startStaking: not deployed');
```

Comment:
I would restrict this to only when the rewards pool is over

Reply:
This is mainly for accidental token transfer. If the rewards pool is over there could be some users that haven't withdrawn their tokens, and we could pull them.